test/epgsql_replication_SUITE.erl

```diff
@@ -75,6 +77,37 @@ replication_async_active_n_ssl(Config) ->
     noop.
 -endif.
 
+two_replications_on_same_slot(Config) ->
+  Module = ?config(module, Config),
+  User = "epgsql_test_replication",
+  Parent = self(),
+  epgsql_ct:with_connection(
+    Config,
+    fun(C) ->
+        create_replication_slot(Config, C),
+        Res1 = Module:start_replication(C, "epgsql_test", Parent, {C, Parent}, "0/0"),
+        ?assertEqual(ok, Res1),
+        spawn(
+          fun() ->
+              %% This connection will be terminated due to an unexpected message:
+              %%   EmptyQueryResponse (B), Byte1('I'), Int32(4)
+              %% https://www.postgresql.org/docs/current/protocol-message-formats.html
+              process_flag(trap_exit, true),
+              C2 = epgsql_ct:connect(Config, User, [{replication, "database"}]),
+              Res2 = Module:start_replication(C2, "epgsql_test", self(), {C2, self()}, "0/0"),
```

Comment:
I'm not a big expert in replication protocol... Does it mean it's not allowed to have two replications on the same slot?
May you please add some kind of a diagram of what heppens when someone attempts to launch a 2nd replication on the same slot?

The 2nd attempt message exchange would look like this?

```
> SimpleQuery "START_REPLICATION ..."
< CopyBothResponse
< EmptyQueryResponse
```

Reply:
I just can't find `EmptyQueryResponse` anywhere in the docs in context of the replication https://www.postgresql.org/docs/15/protocol-replication.html

Reply:
> Does it mean it's not allowed to have two replications on the same slot?

Yes, only one process can occupy a slot.

> The 2nd attempt message exchange would look like this?
> 
> ```
> > SimpleQuery "START_REPLICATION ..."
> < CopyBothResponse
> < EmptyQueryResponse
> ```

Seems like it is. I don't know why it happens, unfortunately. I have tried to find out by googling and monitoring a connection quickly, but I didn't find the reason. Maybe it is a bug in the protocol...

Reply:
My mistake, the correct flow is the following:
```
> SimpleQuery "START_REPLICATION ..."
< CopyBothResponse
< ReadyForQuery
```